File: verbNet/src/main/java/org/sqlunet/verbnet/sql/VnFrameQueryFromClassIdAndSense.java
/*
 * Copyright (c) 2022. Bernard Bou
 */

package org.sqlunet.verbnet.sql;

import android.database.sqlite.SQLiteDatabase;

import org.sqlunet.sql.DBQuery;

import java.sql.Types;

import androidx.annotation.Nullable;

/**
 * VerbNet frame query from sense
 *
 * @author <a href="mailto:1313ou@gmail.com">Bernard Bou</a>
 */
class VnFrameQueryFromClassIdAndSense extends DBQuery
{
	/**
	 * <code>QUERY</code> is the SQL statement
	 */
	static private final String QUERY = SqLiteDialect.VerbNetFramesQueryFromClassIdAndSense;

	/**
	 * Constructor
	 *
	 * @param connection connection
	 * @param classId    target class id
	 * @param wordId     target word id
	 * @param synsetId   target synset id
	 */
	@SuppressWarnings("boxing")
	public VnFrameQueryFromClassIdAndSense(final SQLiteDatabase connection, final long classId, final long wordId, @Nullable final Long synsetId)
	{
		super(connection, VnFrameQueryFromClassIdAndSense.QUERY);
		setParams(classId, wordId);
		if (synsetId != null)
		{
			this.statement.setLong(2, synsetId);
		}
		else
		{
			this.statement.setNull(2, Types.DECIMAL);
		}
	}

	/**
	 * Get the frame id from the result set
	 *
	 * @return the frame id from the result set
	 */
	public long getFrameId()
	{
		assert this.cursor != null;
		return this.cursor.getLong(0);
	}

	/**
	 * Get the frame number from the result set
	 *
	 * @return the frame number from the result set
	 */
	public String getNumber()
	{
		assert this.cursor != null;
		return this.cursor.getString(1);
	}

	/**
	 * Get the frame xtag from the result set
	 *
	 * @return the frame xtag from the result set
	 */
	public String getXTag()
	{
		assert this.cursor != null;
		return this.cursor.getString(2);
	}

	/**
	 * Get the frame major description from the result set
	 *
	 * @return the frame major description from the result set
	 */
	public String getDescription1()
	{
		assert this.cursor != null;
		return this.cursor.getString(3);
	}

	/**
	 * Get the frame minor description from the result set
	 *
	 * @return the frame minor description from the result set
	 */
	public String getDescription2()
	{
		assert this.cursor != null;
		return this.cursor.getString(4);
	}

	/**
	 * Get the frame syntax from the result set
	 *
	 * @return the frame syntax from the result set
	 */
	public String getSyntax()
	{
		assert this.cursor != null;
		return this.cursor.getString(5);
	}

	/**
	 * Get the frame semantics from the result set
	 *
	 * @return the frame semantics from the result set
	 */
	public String getSemantics()
	{
		assert this.cursor != null;
		return this.cursor.getString(6);
	}

	/**
	 * Get the frame example from the result set
	 *
	 * @return the frame example from the result set
	 */
	public String getExample()
	{
		assert this.cursor != null;
		return this.cursor.getString(7);
	}

	/**
	 * Get the frame quality from the result set
	 *
	 * @return the frame quality from the result set
	 */
	public int getQuality()
	{
		assert this.cursor != null;
		return this.cursor.getInt(8);
	}

	/**
	 * Get synset-specific flag from the result set
	 *
	 * @return the synset-specific flag from the result set
	 */
	public boolean getSynsetSpecific()
	{
		assert this.cursor != null;
		final int result = this.cursor.getInt(9);
		return result == 0;
	}
}

File: verbNet/src/main/java/org/sqlunet/verbnet/sql/VnFrameQueryFromClassIdAndSense.kt
/*
 * Copyright (c) 2023. Bernard Bou
 */
package org.sqlunet.verbnet.sql

import android.database.sqlite.SQLiteDatabase
import org.sqlunet.sql.DBQuery
import java.sql.Types

/**
 * VerbNet frame query from sense
 *
 * @param connection connection
 * @param classId    target class id
 * @param wordId     target word id
 * @param synsetId   target synset id
 *
 * @author [Bernard Bou](mailto:1313ou@gmail.com)
 */
internal class VnFrameQueryFromClassIdAndSense(connection: SQLiteDatabase, classId: Long, wordId: Long, synsetId: Long?) : DBQuery(connection, QUERY) {

    init {
        setParams(classId, wordId)
        if (synsetId != null) {
            statement.setLong(2, synsetId)
        } else {
            statement.setNull(2, Types.DECIMAL)
        }
    }

    /**
     * Frame id from the result set
     */
    @Suppress("unused")
    val frameId: Long
        get() = cursor!!.getLong(0)

    /**
     * Frame number from the result set
     */
    val number: String
        get() = cursor!!.getString(1)

    /**
     * Frame xtag from the result set
     */
    val xTag: String
        get() = cursor!!.getString(2)

    /**
     * Frame major description from the result set
     */
    val description1: String
        get() = cursor!!.getString(3)

    /**
     * Frame minor description from the result set
     */
    val description2: String
        get() = cursor!!.getString(4)

    /**
     * Frame syntax from the result set
     */
    val syntax: String
        get() = cursor!!.getString(5)

    /**
     * Frame semantics from the result set
     */
    val semantics: String
        get() = cursor!!.getString(6)

    /**
     * Frame example from the result set
     */
    val example: String
        get() = cursor!!.getString(7)

    /**
     * Frame quality from the result set
     */
    val quality: Int
        get() = cursor!!.getInt(8)

    /**
     * Synset-specific flag from the result set
     */
    @Suppress("unused")
    val synsetSpecific: Boolean
        get() =
            cursor!!.getInt(9) == 0

    companion object {

        /**
         * `QUERY` is the SQL statement
         */
        private const val QUERY = SqLiteDialect.VerbNetFramesQueryFromClassIdAndSense
    }
}
